client/src/components/DetailsEvent/index.js

```diff
@@ -1,13 +1,180 @@
-import React from 'react'
-import Header from '../Header/header';
-class DetailsEvent extends React.Component {
+import React, { Component } from "react";
+import NavBar from "../NavBar";
+import SideBar from "../SideBar";
+import axios from "axios";
+
+class DetailsEvent extends Component {
+  state = {
+    id: this.props.history.location.event.id,
+    status: this.props.history.location.event.status,
+    loading: false
+  };
+
+  componentDidMount() {
+    axios
+      .get(`/event/${this.state.id}`)
+      .then(data => {
+        this.setState({
+          event_details: data.data.event,
+          coffee_details: data.data.coffee,
+          lunch_details: data.data.lunch,
+          equipment_details: data.data.equipment,
+          loading: true
+        });
+      })
+      .catch(() => {
+        const { history } = this.props;
+        history.push("/error");
+      });
+  }
+
+  handleSubmitForm = event => {
+    event.preventDefault();
+    const { history } = this.props;
+
+    if (this.state.status === 0) {
+      axios
+        .put(`/event/${this.state.id}`)
+        .then(data => {
+          history.push("/events");
+        })
+        .catch(() => {
+          history.push("/error");
+        });
+    } else {
+      history.push("/events");
+    }
+  };
+
+  cancel = () => {
+    const { history } = this.props;
+    history.push("/events");
+  };
+
   render() {
-    return (
-      <div>
-      <Header />
-      <h1>Event Details </h1>
-      </div>
-    )
+    if (this.state.loading === true) {
+      const {
+        title = "event-title",
+        start_date = "",
+        end_date = "",
+        org_name = "",
+        price = "",
+        capacity = "",
+        note = ""
+      } = this.state.event_details || {};
+
+      const { coffee_note = "", coffee_price = "", coffee_time = "" } = {
+        ...this.state.coffee_details
+      };
+
+      const { lunch_note = "", lunch_price = "", lunch_time = "" } =
+        this.state.lunch_details || {};
+
+      const { equipment_price = "", equipment_note = "" } =
+        this.state.equipment_details || {};
+
+      let start_ddmmyyyy = new Date(start_date);
+
+      start_ddmmyyyy =
+        start_ddmmyyyy.getDate() +
+        "-" +
+        start_ddmmyyyy.getMonth() +
+        "-" +
+        start_ddmmyyyy.getFullYear();
+
+      let end_ddmmyyyy = new Date(end_date);
+
+      end_ddmmyyyy =
+        end_ddmmyyyy.getDate() +
+        "-" +
+        end_ddmmyyyy.getMonth() +
+        "-" +
+        end_ddmmyyyy.getFullYear();
+      return (
+        <div className="page">
+          <div>
+            <NavBar />
+          </div>
+          <div className="both">
+            <SideBar />
+            <div className="detailsEvent">
+              <form onSubmit={this.handleSubmitForm}>
+                <h2 className="h2">General</h2>
+                <div className="labels_container">
+                  <label className="title">Event title:</label>
```

Comment:
Labels with no inputs? Unless you can edit these Maybe both should just be `div`s? Or am I missing something

Reply:
yeah we have to change this, will be div and p I guess